ERCS/erc-7683.md

```diff
@@ -126,29 +129,29 @@ struct Output {
 	/// @dev The destination chain for this output
 	uint64 chainId;
 }
-```
-
-### Initiated event
 
-A compliant `Initiated` event MUST adhere to the following abi:
-
-```solidity
-/// @title FillInstructions type
+/// @title FillInstruction type
 /// @notice Instructions to parameterize each leg of the fill
 /// @dev Provides all the origin-generated information required to produce a valid fill leg
-struct FillInstructions {
+struct FillInstruction {
 	/// @dev The contract address that the order is meant to be settled by
 	uint64 destinationChainId;
 	/// @dev The contract address that the order is meant to be filled on
 	bytes32 destinationSettler;
 	/// @dev The data generated on the origin chain needed by the destinationSettler to process the fill
 	bytes originData;
 }
+```
+
+### Initiate event
+
+A compliant `Initiate` event MUST adhere to the following abi:
 
+```solidity
 /// @notice Signals that an order has been initiated
 /// @param orderId a unique order identifier within this settlement system
-/// @param fillInstructions fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
-event Initiated(bytes32 indexed orderId, FillInstructions[] fillInstructions);
+/// @param fillInstructions instructions to parameterize each fill leg, with each element representing a single leg
+event Initiate(bytes32 indexed orderId, ResolvedCrossChainOrder resolvedOrder);
```

Comment:
just flagging that this will be a pretty beefy event; I'd guess that having this whole struct will add easily 5k to 10k gas

Reply:
Yep, agreed. Do you see a way to either reduce the size or get rid of the event without hurting filler UX?

Reply:
I'd still contend that if the filler is the one calling this function, they already know what is going to be resolved here; I think we should either make the event optional or only emit the truly dynamic components of the resolved order

Reply:
Also, if a filler _does_ need the event, they could trigger the initiation from a contract that then takes the returndata and uses it to encode and emit the event

Reply:
in the non-gasless flow, I agree that either the event needs be emitted or that the `resolve` return value needs to be consistent!